### PR TITLE
Add CNI example

### PR DESCRIPTION
**Pull request by adleong**
Cribbed from https://github.com/apprenda/kismatic/blob/master/ansible/roles/addon-linkerd/templates/linkerd.yml

### CODE REVIEW

File: k8s-daemonset/k8s/linkerd-cni.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      host: localhost
+      port: 8001
+
+    telemetry:
+    - kind: io.l5d.prometheus
+    - kind: io.l5d.recentRequests
+      sampleRate: 0.25
+
+    usage:
+      orgId: linkerd-examples-daemonset
```
Comment (klingerf) [inline k8s-daemonset/k8s/linkerd-cni.yml]: Can you set this to `linkerd-examples-daemonset-cni`? That more consistent with the rest of the configs in this directory.

File: k8s-daemonset/k8s/linkerd-cni.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        configMap:
+          name: "l5d-config"
+      containers:
+      - name: l5d
+        image: buoyantio/linkerd:0.9.1
+        env:
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
```
Comment (klingerf) [inline k8s-daemonset/k8s/linkerd-cni.yml]: Do you also have the set the NODE_NAME env variable here?

Comment (adleong) [inline k8s-daemonset/k8s/linkerd-cni.yml]: yes, good catch!

**Comment by siggy** [comment]
would this config work in GKE? if so, should we just recommend this way all the time?

**Comment by adleong** [comment]
I think it would, but the semantics are slightly different.  This runs linkerd on the host network which (for some reason) is required in order to use `hostPort` in CNI setups.  Our default config is able to use hostPort without running linkerd on the host network.

**Pull request merged**